projects/aca-content/viewer/src/lib/services/viewer.service.ts

```diff
@@ -0,0 +1,229 @@
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ObjectUtils, UserPreferencesService } from '@alfresco/adf-core';
+import { FavoritePaging, Node, NodePaging, SearchRequest, ResultSetPaging, SharedLink, SharedLinkPaging } from '@alfresco/js-api';
+import { Injectable } from '@angular/core';
+import { ContentApiService } from '@alfresco/aca-shared';
+
+interface adjacentFiles {
+  left: string;
+  right: string;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ViewerService {
+  constructor(private preferences: UserPreferencesService, private contentApi: ContentApiService) {}
+
+  recentFileFilters = [
+    'TYPE:"content"',
+    '-PATH:"//cm:wiki/*"',
+    '-TYPE:"app:filelink"',
+    '-TYPE:"fm:post"',
+    '-TYPE:"cm:thumbnail"',
+    '-TYPE:"cm:failedThumbnail"',
+    '-TYPE:"cm:rating"',
+    '-TYPE:"dl:dataList"',
+    '-TYPE:"dl:todoList"',
+    '-TYPE:"dl:issue"',
+    '-TYPE:"dl:contact"',
+    '-TYPE:"dl:eventAgenda"',
+    '-TYPE:"dl:event"',
+    '-TYPE:"dl:task"',
+    '-TYPE:"dl:simpletask"',
+    '-TYPE:"dl:meetingAgenda"',
+    '-TYPE:"dl:location"',
+    '-TYPE:"fm:topic"',
+    '-TYPE:"fm:post"',
+    '-TYPE:"ia:calendarEvent"',
+    '-TYPE:"lnk:link"'
+  ];
+
+  /**
+   * Retrieves nearest node information for the given node and folder.
+   *
+   * @param nodeId Unique identifier of the document node
+   * @param folderId Unique identifier of the containing folder node.
+   * @param source Data source name. Allowed values are: personal-files, libraries, favorites, shared, recent-files.
+   */
+  async getNearestNodes(nodeId: string, folderId: string, source: string): Promise<adjacentFiles> {
+    const empty: adjacentFiles = {
+      left: null,
+      right: null
+    };
+
+    if (nodeId && folderId) {
+      try {
+        const ids = await this.getFileIds(source, folderId);
+        const idx = ids.indexOf(nodeId);
+
+        if (idx >= 0) {
+          return {
+            left: ids[idx - 1] || null,
+            right: ids[idx + 1] || null
+          };
+        }
+      } catch {}
+    }
+    return empty;
+  }
+
+  /**
+   * Retrieves a list of node identifiers for the folder and data source.
+   *
+   * @param source Data source name. Allowed values are: personal-files, libraries, favorites, shared, recent-files.
+   * @param folderId Optional parameter containing folder node identifier for 'personal-files' and 'libraries' sources.
+   */
+  async getFileIds(source: string, folderId?: string): Promise<string[]> {
```

Comment:
similarly like above - maybe we can limit type of source to only these allowed values instead of general string?